src/frontend/src/components/modal.ts

```diff
@@ -0,0 +1,59 @@
+import { html, TemplateResult, render } from "lit-html";
+
+export const createModal = ({
+  slot,
+  id,
+  confirmText = "Ok",
+}: {
+  slot: TemplateResult;
+  id?: string;
+  confirmText?: string;
+}): HTMLDialogElement => {
+  const modalHtml = html`
+    <form method="dialog" class="c-modal__content c-card c-card--modal">
+      <div class="c-modal__inner">
+        <span class="c-modal__close" data-closemodal>&times;</span>
```

Comment:
How about adding the `close` functions here via `@click=${ () => ... }` instead of looking up the elements later on with `querySelector`? If you need a placeholder for the modal before it is created you can use `createRef()` and `Ref`

Reply:
ah good to know

Reply:
addressed this in 83e0a6f9ad571c77710b51ae7e9889a9cab866eb I passed the reference in from the promise since it needs to resolve it. Can you think of a nicer way to do that?